spec/controllers/catalog_controller_spec.rb

```diff
@@ -1,46 +1,92 @@
 require 'rails_helper'
-
 RSpec.describe CatalogController, type: :controller do
-  describe 'POST #add_preserved_object' do
+  describe 'POST #create' do
+    let(:size) { 2342 }
+    let(:ver) { 3 }
+    let(:endpoint_name) { 'fixture_sr1' }
+    let(:druid) { 'bj102hs9687' }
+
     context 'with valid params' do
-      it 'saves the object with the expected field values' do
-        allow(Stanford::StorageServices).to receive(:current_version).and_return(6)
-        allow(subject).to receive(:moab_size).and_return(555)
-        id = 'ab123cd45678'
+      let(:pres_copy) { PreservedCopy.first }
+      let(:pres_obj) { PreservedObject.first }
+
+      before do
+        post :create, params: { druid: druid, incoming_version: ver, incoming_size: size, endpoint_name: endpoint_name }
+      end
+
+      it 'saves PreservedObject and PreservedCopy in db' do
+        expect(PreservedObject).to exist
```

Comment:
is this the class or the object?   should it be `PreservedObject.find(druid)` instead of `PreservedObject`?

Reply:
From RSpec docs:
```
  expect(obj).to exist # passes if obj.exist? or obj.exists?
```

When I try:

```
it 'saves PreservedObject and PreservedCopy in db' do
  pres_obj = PreservedObject.find_by(druid: druid)
  expect(pres_obj).to exist
end
```
it fails with:
```
Failure/Error: expect(pres_obj).to exist
       expected #<PreservedObject id: 1, druid: "bj102hs9687", current_version: 3, created_at: "2018-04-05 16:30:46", updated_at: "2018-04-05 16:30:46", preservation_policy_id: 1> to exist but it does not respond to either `exist?` or `exists?`
```

However; I did try in the console that when there are no objects in PreservedObject or PreservedCopy table, ` PreservedCopy.exists?` and `PreservedObject.exists?` both return false.

```
input> PreservedObject.all
=>   PreservedObject Load (3.1ms)  SELECT "preserved_objects".* FROM "preserved_objects"
[]
input> PreservedObject.exists?
  PreservedObject Exists (0.4ms)  SELECT  1 AS one FROM "preserved_objects" LIMIT $1  [["LIMIT", 1]]
=> false
```

and then when I create 1 object in the PreservedCopy and PreservedObject table I get:

```
input> PreservedObject.exists?
  PreservedObject Exists (0.4ms)  SELECT  1 AS one FROM "preserved_objects" LIMIT $1  [["LIMIT", 1]]
=> true
```

Reply:
er ... wouldn't you do something like

```
it 'saves PreservedObject and PreservedCopy in db' do
  pres_obj = PreservedObject.find_by(druid: druid)
  expect(pres_obj).to be_an_instance of PreservedObject
end
```

Basically, you are proving the specific object is there.  Does find_by return nil if it is NOT there?

Note that `PreservedObject.exist?` says nothing about WHICH object -- it just says there is an entry in the table.

Reply:
Yes find_by returns `nil` if it is NOT there.

True - I'll change it to yours. I like it better.